crates/anvil/src/config.rs

```diff
@@ -997,9 +1014,55 @@ impl NodeConfig {
                 .expect("Failed to establish provider to fork url"),
         );
 
-        let (fork_block_number, fork_chain_id) = if let Some(fork_block_number) =
-            self.fork_block_number
+        let (fork_block_number, fork_chain_id, force_transactions) = if let Some(fork_choice) =
+            &self.fork_choice
         {
+            let (fork_block_number, force_transactions) = match fork_choice {
+                ForkChoice::Block(block_number) => (block_number.to_owned(), Vec::new()),
+                ForkChoice::Transaction(transaction_hash) => {
+                    // Determine the block that this transaction was mined in
```

Comment:
could we extract this scope to a new function that returns a `Result`?

then we only need to call expect/unwrap once

Reply:
updated